airbyte-webapp/src/components/base/Input/Input.tsx

```diff
@@ -83,46 +83,82 @@ const VisibilityButton = styled(Button)`
   border: none;
 `;
 
-const Input: React.FC<InputProps> = ({ defaultFocus = false, onFocus, onBlur, ...props }) => {
+const Input: React.FC<InputProps> = ({ defaultFocus = false, ...props }) => {
   const { formatMessage } = useIntl();
+
   const inputRef = useRef<HTMLInputElement | null>(null);
-  const [isContentVisible, setIsContentVisible] = useToggle(false);
+  const buttonRef = useRef<HTMLButtonElement | null>(null);
+  const inputSelectionStartRef = useRef<number | null>(null);
+
+  const [isContentVisible, toggleIsContentVisible] = useToggle(false);
   const [focused, setFocused] = useState(false);
 
   const isPassword = props.type === "password";
   const isVisibilityButtonVisible = isPassword && !props.disabled;
   const type = isPassword ? (isContentVisible ? "text" : "password") : props.type;
 
+  const focusOnInputElement = useCallback(() => {
+    if (!inputRef.current) {
+      return;
+    }
+
+    const { current: element } = inputRef;
+    const selectionStart = inputSelectionStartRef.current ?? inputRef.current?.value.length;
+
+    element.focus();
+
+    if (selectionStart) {
+      // Update input cursor position to where it was before
+      window.setTimeout(() => {
+        element.setSelectionRange(selectionStart, selectionStart);
+      }, 0);
+    }
+  }, []);
+
   useEffect(() => {
-    if (defaultFocus && inputRef.current !== null) {
-      inputRef.current.focus();
+    if (defaultFocus) {
+      focusOnInputElement();
+    }
+  }, [defaultFocus, focusOnInputElement]);
```

Comment:
I feel like `defaultFocus` could just be `autofocus={defaultFocus}` on the `InputComponent` below.
It's only set in one place and I don't believe it changes.